packages/odyssey-react-mui/src/PasswordField.tsx

```diff
@@ -116,9 +117,26 @@ const PasswordField = forwardRef<HTMLInputElement, PasswordFieldProps>(
     },
     ref,
   ) => {
+    const TEXT_TYPE_TIMEOUT = 30000;
     const { t } = useTranslation();
     const [inputType, setInputType] = useState("password");
 
+    useEffect(() => {
+      let timeout: NodeJS.Timeout;
+
+      // If current inputType is text (cleartext password),
+      // for security set a 30-second timeout to toggle back to password
+      if (inputType === "text") {
+        timeout = setTimeout(() => {
+          setInputType("password");
+        }, TEXT_TYPE_TIMEOUT);
+      }
+
+      return () => {
+        clearTimeout(timeout);
+      };
+    }, [inputType]);
```

Comment:
```suggestion
    useEffect(() => {
      let timeoutId: NodeJS.Timeout;

      // If current inputType is text (cleartext password),
      // for security set a 30-second timeout to toggle back to password
      if (inputType === "text") {
        timeoutId = setTimeout(() => {
          setInputType("password");
        }, PASSWORD_VISIBILITY_TIMEOUT);
      }

      return () => {
        clearTimeout(timeoutId);
      };
    }, [inputType]);
```

Reply:
should `timeoutId` be a `ref`?

Reply:
If I understand it correctly, we only need to use a ref if we are passing the timeout ref to a different component (i.e. child component) to be cleared, given this is the same component and within a useEffect I don't believe it is needed, but please let me know if I misunderstood 🙏

Reply:
Yeah, you are probably right. I think because the timeout is only used within this hook, `let` is probably fine. I'm just so used to creating them as `ref` I just do that by default.